main.go

```diff
@@ -103,42 +109,39 @@ func main() {
 	}
 }
 
-type conf struct {
-	konsAPIKey      string
-	konsDNSEndPoint string
-	elbName         string
-	awsAccessKey    string
-	awsSecretKey    string
-	awsRegion       string
-}
-
-func elbDNSName(c *conf) string {
-	//weirdness around how aws handles credentials
-	os.Setenv("AWS_ACCESS_KEY_ID", c.awsAccessKey)
-	os.Setenv("AWS_SECRET_ACCESS_KEY", c.awsSecretKey)
-
-	svc := elb.New(
-		session.New(
-			&aws.Config{
-				Region:      aws.String(c.awsRegion),
-				Credentials: credentials.NewEnvCredentials(),
-			},
-		),
-	)
-
-	params := &elb.DescribeLoadBalancersInput{
-		LoadBalancerNames: []*string{
-			aws.String(c.elbName), // Required
-		},
+func getKubeElbDnsCname(conf *conf) string {
+	// creates the in-cluster config
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Fatalf("ERROR - Could not get the K8s cluster config: [%v]", err)
+	}
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Fatalf("ERROR - Could not get the client for K8s: [%v]", err)
 	}
 
-	resp, err := svc.DescribeLoadBalancers(params)
-
+	lbService, err := clientset.Core().Services(conf.kubeLbServiceNamespace).Get(conf.kubeLbService)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatalf("ERROR - Could not get the K8S LB service '%s'. Reason: [%v]", conf.kubeLbService, err)
 	}
 
-	return *resp.LoadBalancerDescriptions[0].DNSName
+	if (len(lbService.Status.LoadBalancer.Ingress) == 0) {
+		log.Fatalf("ERROR - No ingress address found for LB service '%s'", conf.kubeLbService)
```

Comment:
Yep, that's exactly the behavior I want, as I start this as Job that will keep creating the pod until it won't fail.